ext/serializer.c

```diff
@@ -1268,6 +1295,23 @@ void ddtrace_serialize_span_to_array(ddtrace_span_data *span, zval *array) {
         _add_assoc_zval_copy(el, "type", &prop_type_as_string);
     }
 
+    if (span_type) {
+        zend_hash_str_del(meta, ZEND_STRL("span.type"));
```

Comment:
So many strings repeated everywhere :p Why not constants? (comment can clearly be ignored as not particular to this PR)

Reply:
Do you mean that you'd want strings like "span.type", "service.name", "_dd1.sr.eausr", "_dd.span_sampling.rule_rate" (to name a few) to be constants? (Pretty much like the [Tag](https://github.com/DataDog/dd-trace-php/blob/a1f35f6952cff87f5e4c31ee5534eee76d91bcac/src/api/Tag.php#L5) class in PHP?)

Reply:
I'm not a particularly big fan of constants which are identical to the value they represent (`#define SPAN_KIND "span.kind"`) or such.
For strings which are more complex, sure.

I prefer optimizing the code for readability - I read the code, I immediately see the bare full name of the value.
At least, as long as one is testing the behaviours, it will be caught in testing, should it be mistyped. Constants are mostly a tool to immediately see the issue upon compilation instead of testing. But I think the tradeoff isn't worth it.